Strange "Returning Any" warning

`mypy --warn-return-any` (ver. 0.550) gives a strange "Returning Any from function ..." warning for the following code:
```
def f(a: 'A') -> int:
    if a._x == 2:
        return 2
    n = 1
    return n

class A:
    def __init__(self) -> None:
        self._x = 2
```
```
foo.py:5: warning: Returning Any from function declared to return "int"
```
Combining `n = 1` `return n` into one statement `return 1` or putting the definition of `A` before `f` makes the warning disappear.

--- a/mypy/checker.py
+++ b/mypy/checker.py
@@ -2133,7 +2133,7 @@ class TypeChecker(NodeVisitor[None], CheckerPluginInterface):
                 if isinstance(typ, AnyType):
                     # (Unless you asked to be warned in that case, and the
                     # function is not declared to return Any)
-                    if (self.options.warn_return_any and
+                    if (self.options.warn_return_any and not self.current_node_deferred and
                             not is_proper_subtype(AnyType(TypeOfAny.special_form), return_type)):
                         self.msg.incorrectly_returning_any(return_type, s)
                     return

--- a/test-data/unit/check-warnings.test
+++ b/test-data/unit/check-warnings.test
@@ -209,5 +209,3 @@ def foo(a1: A) -> int:
 class A:
     def __init__(self, x: int) -> None:
         self._x = x
-
-

--- a/test-data/unit/check-warnings.test
+++ b/test-data/unit/check-warnings.test
@@ -197,3 +197,17 @@ class Test(object):
             return None
 [builtins fixtures/list.pyi]
 [out]
+
+[case testReturnAnyDeferred]
+# flags: --warn-return-any
+def foo(a1: A) -> int:
+    if a1._x:
+        return 1
+    n = 1
+    return n
+
+class A:
+    def __init__(self, x: int) -> None:
+        self._x = x
+
+

